### PR TITLE
test(message_edit_post_history): New test case for edit post history

### PR DESCRIPTION
**Pull request by jgilliam17**
#### Summary

Adding new test case for message edit post history. Contributor is adding Cypress test.

Jira ticket: https://mattermost.atlassian.net/browse/MM-45494

#### Is the feature in stable branch or under review?
- [] Feature in stable branch (master, main)?
- [x] Feature still under review? If yes, list all related pull requests for reference.
Webapp PR: https://github.com/mattermost/mattermost-webapp/pull/11069
Server PR: https://github.com/mattermost/mattermost-server/pull/20945
Server PR: https://github.com/mattermost/mattermost-server/pull/21601
#### Test server

https://mattermost-server-pr-21601.test.mattermost.cloud/
#### Test client

- [x] Browser
- [] Mobile App (version, OS)
- [x] Desktop App (version, OS)

<a href="https://gitpod.io/#https://github.com/mattermost/mattermost-test-management/pull/62"></a>

### CODE REVIEW

File: data/test-cases/channels/messaging/message_edit_post_history.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+**Step 1**
+
+Show and restore older versions of a message
+
+1. Login as test user and visit off-topic
+    1. Validate if the channel has been opened
+2. Post a message "This is a sample message"
+3. Edit message "This is an edited sample message"
+4. Edit message again to "This is the final version of the sample message"
+5. Open edit history by clicking on edited post link
```
Comment (saturninoabril) [inline data/test-cases/channels/messaging/message_edit_post_history.md:44]: re: "edited post link" - Not sure if "link" is clear to anyone reading it first. Maybe button? Could be:

```suggestion
5. Open edit history by clicking "`<pencil icon>` Edited" button of the post
```

File: data/test-cases/channels/messaging/message_edit_post_history.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+User can restore previous version of the message from Edit History
+
+---
+
+**Step 2**
+
+Show, restore older versions of a message and click undo in toast
+
+1. Get the last post id
```
Comment (saturninoabril) [inline data/test-cases/channels/messaging/message_edit_post_history.md:62]: "id" is not visible to actual user. Perhaps,
```suggestion
1. Get the last post
```

File: data/test-cases/channels/messaging/message_edit_post_history.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+2. Click the restore button on the first version of the message
+3. COnfirm the resttore
```
Comment (saturninoabril) [inline data/test-cases/channels/messaging/message_edit_post_history.md:64]:
```suggestion
3. Confirm the restore
```

File: data/test-cases/channels/messaging/message_edit_post_history.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+4. Wait for the message to be updated 
+5. Click the undo button on the toast
+6. Wait for the message to be updated 
+7. Check if the messsage has been updated to the correct version "This is the final version of the sample message"
```
Comment (saturninoabril) [inline data/test-cases/channels/messaging/message_edit_post_history.md:68]:
```suggestion
7. Check if the message has been updated to the correct version "This is the final version of the sample message"
```

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @jgilliam17!

Is there an existing test that exercise permission on who can/cannot edit a post? If not, may be something to add? Not necessarily blocking here or for [MM-45494](https://mattermost.atlassian.net/browse/MM-45494).

File: data/test-cases/channels/messaging/message_edit_post_history.md (hunk 2 above)
Comment (jgilliam17) [reply data/test-cases/channels/messaging/message_edit_post_history.md:62]: Great point! Thanks

**Comment by jgilliam17** [comment]
> Thanks @jgilliam17!
> 
> Is there an existing test that exercise permission on who can/cannot edit a post? If not, may be something to add? Not necessarily blocking here or for [MM-45494](https://mattermost.atlassian.net/browse/MM-45494).

I'll have to check on this. I'll mention in the webapp or server PR

**Review by stevemudie — APPROVED** [review]

**Comment by jgilliam17** [comment]
@saturninoabril Can you sync with Zephyr and share test case ID please? Thanks 🙂

**Comment by saturninoabril** [comment]
Hello @jgilliam17, done, MM-T5381.

**Comment by jgilliam17** [comment]
Great - thanks Saturn!

**Pull request merged**